memberships/forms.py

```diff
@@ -19,17 +21,52 @@ class RegistrationForm(forms.Form):
     donation = forms.DecimalField(min_value=0, decimal_places=2, required=False)
 
     def clean_birth_date(self, *args, **kwargs):
-        from funky_time import is_18, date_to_datetime
+        from funky_time import is_younger_than, is_older_than, date_to_datetime
 
         birth_date = date_to_datetime(self.cleaned_data.get("birth_date"))
 
-        if is_18(birth_date) is False:
+        if is_younger_than(0, birth_date):
+            raise forms.ValidationError("Unless you are a Time Lord, please enter a date in the past.")
+
+        elif is_younger_than(18, birth_date):
             raise forms.ValidationError(
                 "Thanks for your interest in joining Geek.Zone! We're pumped that you"
                 " want to become an official epic Geek, however, as you are under 18 we"
                 " need to speak to your parent or guardian. Please ask them to email"
                 " trustees@geek.zone to request membership on your behalf. Thanks!"
             )
-        # FIXME JDG in the future, we should put messages like these in the admin area for future changes
+
+        elif is_older_than(130, birth_date):
+            raise forms.ValidationError("Nobody has ever lived that long! Please check your birthdate.")
+
+        # FIXME JDG in the future, messages and limits like these should be admin user configurable
 
         return birth_date
+
+
+class MemberSettingsForm(ModelForm):
+        class Meta:
+            model = Member
+            fields = '__all__'
+            exclude = [
+                'stripe_customer_id',
+                'email',
+                'user',
+                'constitution_agreed'
+            ]  # JDG Should also exclude renewal date once we have it
+            widgets = {'birth_date': DateInput()}
+
+
+class MemberDetailsForm(ModelForm):
+    class Meta:
+        model = Member
+        fields = '__all__'
+        exclude = [
+            'stripe_customer_id',
+            'email',
+            'user',
+            'constitution_agreed',
+            'profile_image'
+        ]
+        widgets = {'birth_date': DateInput()}
```

Comment:
If this is just for rendering out the current user details, do we need this widgets section?

Reply:
Indeed we don't; it is required in the `MemberSettingsForm` to ensure that this field is a date picker, but as we have disabled all the fields in `MemberDetailsForm` it does not matter.